### PR TITLE
[TR] PIM-5666: Fix number comparator

### PR DESCRIPTION
**Pull request by willy-ahva**
Save a new metric product value filled with '0' on a product, save the new attribute but not the value "0".

In the NumberComparator we were comparing `(float) 0` the new value with `(float) null` the value from db. But `(float) null` is equal to `0`, PHP is magic \o/

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | yes
| Added Behats                      | yes
| Changelog updated                 | yes
| Review and 2 GTM                  |
| Micro Demo to the PO (Story only) |
| Migration script                  |
| Tech Doc                          |

EE ODM = http://core-ci.akeneo.com/job/dci_behat/3882/ 1 random

### CODE REVIEW

**Comment by juliensnz** [comment]
GTM

File: CHANGELOG-1.5.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 ## Bug fixes
 
 - PIM-5643: Fix default system locale
+- PIM-5666: Fix NumberComparator. Save a product with a new metric custom attribute filled with '0', well save the value.
```
Comment (jjanvier) [inline CHANGELOG-1.5.md]: I don't understand the sentence ^^

File: features/product/create_product_and_save_added_attributes.feature [hunk 2]
```diff
@@ -0,0 +1,36 @@
+@javascript
+Feature: Create product and save a new product value
+  In order to enrich a new product
+  As a product manager
+  I need to be able to create a product, add attributes and save
+
+  Scenario: Successfully create a product, fill product values with 0 and save
```
Comment (jjanvier) [inline features/product/create_product_and_save_added_attributes.feature]: fill in

File: features/product/create_product_and_save_added_attributes.feature [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    Given a "footwear" catalog configuration
+    And the following products:
```
Comment (jjanvier) [inline features/product/create_product_and_save_added_attributes.feature]: why do you need sandals and then to create a new product??

File: features/product/create_product_and_save_added_attributes.feature [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+      | sku     | family  |
+      | sandals | sandals |
+    And the following attributes:
+      | code      | label     | type   | group     |
+      | tmp_price | Tmp Price | prices | marketing |
+    And I am logged in as "Julia"
+    And I am on the products page
+    And I create a new product
+    Then I should see the SKU and Family fields
```
Comment (jjanvier) [inline features/product/create_product_and_save_added_attributes.feature]: not what you have to test here imo

**Comment by jjanvier** [comment]
sorry but I really don't understand the fix
could you explain what you did in the comparator please?

File: CHANGELOG-1.5.md (hunk 1 above)
Comment (SamirBoulil) [inline CHANGELOG-1.5.md]: maybe: 'Fix default product value saved of new custom metric attribute' ?

File: features/product/create_product_and_save_added_attributes.feature (hunk 2 above)
Comment (SamirBoulil) [inline features/product/create_product_and_save_added_attributes.feature]: 'Successfully create a product with product values filled with 0 and save'

**Comment by SamirBoulil** [comment]
gtm small comments

File: features/product/create_product_and_save_added_attributes.feature [hunk 5]
```diff
@@ -0,0 +1,33 @@
+@javascript
+Feature: Create product and save a new product value
+  In order to enrich a new product
+  As a product manager
+  I need to be able to create a product, add attributes and save
+
+  @jira https://akeneo.atlassian.net/browse/PIM-5666
+  Scenario: Successfully create a product, fill in product values with 0 and save
+    Given a "footwear" catalog configuration
+    And the following attributes:
```
Comment (phaseinducer) [inline features/product/create_product_and_save_added_attributes.feature]: No price attribute already in the catalog ?

File: features/product/create_product_and_save_added_attributes.feature [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+      | code      | label     | type   | group     |
+      | tmp_price | Tmp Price | prices | marketing |
+    And I am logged in as "Julia"
+    And I am on the products page
+    And I create a new product
```
Comment (phaseinducer) [inline features/product/create_product_and_save_added_attributes.feature]: Do the bug occur only after product creation ?

**Comment by phaseinducer** [comment]
I don't see the relation between the fix and the behat scenario.

File: features/product/create_product_and_save_added_attributes.feature (hunk 5 above)
Comment (willy-ahva) [inline features/product/create_product_and_save_added_attributes.feature]: There is one but it doesn't accept value under 1.

Comment (phaseinducer) [inline features/product/create_product_and_save_added_attributes.feature]: ok !

File: features/product/create_product_and_save_added_attributes.feature (hunk 6 above)
Comment (willy-ahva) [inline features/product/create_product_and_save_added_attributes.feature]: Nope it occurs too on a already created product, if you add a custom attribute.

**Comment by pierallard** [comment]
Merged in https://github.com/akeneo/pim-community-dev/pull/4410

**Pull request closed without merging**